include/translated-object.php

```diff
@@ -3,176 +3,120 @@
-	 * Left empty for posts.
-	 *
-	 * @var string|null
-	 */
-	protected $object_type;
-
-	/**
-	 * Object type to use when checking capabilities.
-	 *
-	 * @var string
-	 */
-	protected $type;
-
-	/**
-	 * Taxonomy name for the languages.
-	 *
-	 * @var string
-	 */
-	protected $tax_language;
+abstract class PLL_Translated_Object extends PLL_Object_With_Language implements PLL_Translated_Object_Interface {
 
 	/**
 	 * Taxonomy name for the translation groups.
 	 *
 	 * @var string
+	 *
+	 * @phpstan-var non-empty-string
 	 */
 	protected $tax_translations;
 
 	/**
-	 * PLL_Language property name for the term_taxonomy id.
+	 * Object type to use when checking capabilities.
 	 *
 	 * @var string
+	 *
+	 * @phpstan-var non-empty-string
 	 */
-	protected $tax_tt;
+	protected $type;
 
 	/**
 	 * Constructor.
 	 *
 	 * @since 1.8
 	 *
-	 * @param PLL_Model $model Instance of PLL_Model.
+	 * @param PLL_Model $model Instance of `PLL_Model`.
 	 */
-	public function __construct( &$model ) {
-		$this->model = &$model;
+	public function __construct( PLL_Model &$model ) {
+		parent::__construct( $model );
+
+		$this->tax_to_cache[] = $this->tax_translations;
 
 		/*
-		 * Register our taxonomies as soon as possible.
+		 * Register our taxonomy as soon as possible.
 		 * This is early registration, not ready for rewrite rules as $wp_rewrite will be setup later.
 		 */
-		$args = array( 'label' => false, 'public' => false, 'query_var' => false, 'rewrite' => false, '_pll' => true );
-		register_taxonomy( $this->tax_language, $this->object_type, $args );
-		$args['update_count_callback'] = '_update_generic_term_count'; // Count *all* objects to avoid deleting in clean_translations_terms.
-		register_taxonomy( $this->tax_translations, $this->object_type, $args );
+		register_taxonomy(
+			$this->tax_translations,
+			(array) $this->object_type,
+			array(
+				'label'                 => false,
+				'public'                => false,
+				'query_var'             => false,
+				'rewrite'               => false,
+				'_pll'                  => true,
+				'update_count_callback' => '_update_generic_term_count', // Count *all* objects to avoid deleting in clean_translations_terms.
+			)
+		);
 	}
 
 	/**
-	 * Stores the language in the database.
-	 *
-	 * @since 0.6
+	 * Returns the translations group taxonomy name.
 	 *
-	 * @param int                     $id   Object id.
-	 * @param int|string|PLL_Language $lang Language (term_id or slug or object).
-	 * @return void
-	 */
-	abstract public function set_language( $id, $lang );
-
-	/**
-	 * Returns the language of an object.
+	 * @since 3.4
 	 *
-	 * @since 0.1
+	 * @return string
 	 *
-	 * @param int $id Object id.
-	 * @return PLL_Language|false PLL_Language object, false if no language is associated to that object.
+	 * @phpstan-return non-empty-string
 	 */
-	abstract public function get_language( $id );
+	public function get_tax_translations() {
+		return $this->tax_translations;
+	}
```

Comment:
For the record:
- This allows to get the taxonomy name instead of hard-coding it everywhere: this could come handy for future and third-party developments.
- There are similar methods in Polylang for WooCommerce.